### PR TITLE
Changes made to grammar check function

### PR DESCRIPTION
**Pull request by bitanb1999**
Please check the options that you have completed and strike-out the options that do not apply via this pull request:

- [X] a clear title and description to the Pull Request has been provided
 you have read
- [X]  the [Contributing doc](https://github.com/neomatrix369/nlp_profiler/blob/master/CONTRIBUTING.md)
- [X]  the [Developer Guide](https://github.com/neomatrix369/nlp_profiler/blob/master/developer-guide.md)
- [X]  the pull request passes the tests (`./test-coverage "tests slow-tests"`) - this will also be visible via the Code coverage report and CI/CD task on the Pull Request
- [X]  you have performed some kind of smoke test by running your changes in an isolated environment i.e. Docker container, Google Colab, Kaggle, etc...
- ~~[ ]  the notebooks are updated (see notebooks folder, read the [Notebooks](https://github.com/neomatrix369/nlp_profiler/blob/master/.github/notebooks/README.md) docs)~~
- [X]  [CHANGELOG.md](https://github.com/neomatrix369/nlp_profiler/blob/master/CHANGELOG.md) has been updated (please follow the existing format)

### Goal or purpose of the PR

The grammar check function previously used the python language tool, which took significant time to process each text in the textual dataframe and return the output. 

### Changes implemented in the PR

I analyzed the alternatives available in NLP and came across two options: 1. Happy transformers with the hyperparameter tuning of Gramformer( check: https://github.com/PrithivirajDamodaran/Gramformer) and Gingerit package (check: https://github.com/Azd325/gingerit). Gingerit had a throughput time of 34.8 seconds whereas the language tool from python took 41secs to process each text. This seemed to be a huge upgrade.  
Transformers are also a great alternative and did equivalently well but given the constraint of accessing Huggingface every time a text needs to be checked, seems like unnecessary overhead. 
I have made the changes to the requirement file and to the grammar check python file.

### CODE REVIEW

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (neomatrix369) [inline nlp_profiler/high_level_features/grammar_quality_check.py:1]: minor but imp: please apply some formatting - run `black` on your code changes

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (neomatrix369) [inline nlp_profiler/high_level_features/grammar_quality_check.py:42]: good catch

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (neomatrix369) [inline nlp_profiler/high_level_features/grammar_quality_check.py:28]: All of this is good small change, I would keep the parenthese around conditions

The line could have also been written like this:

```python
if not (isinstance(text, str) or text.strip()):
```

See if all the tests pass with the change

**Review by neomatrix369 — CHANGES_REQUESTED** [review]
Overall LGTM - if you can pls address a few comments before we go ahead and merge it,

I'm waiting for the tests to also pass on GitHub actions.

**Comment by neomatrix369** [comment]
Well done with @sourcery-ai improvements

**Comment by neomatrix369** [comment]
Overall really great work doing the analysis and verifying and checking other alternatives to replace the existing slow grammar checker with a better alternative, also shows we can plug-in and play different tooling with little or not too significant changes
.

File: requirements.txt
Comment (neomatrix369) [inline requirements.txt:14]: Safe way to pin the package version number - any drawbacks to keep it open? The good thing maybe that we know it always worked on this version

**Comment by neomatrix369** [comment]
Please also do one last check in https://github.com/neomatrix369/nlp_profiler/blob/master/CONTRIBUTING.md to see if any dependent files need changing i.e. re-running notebooks etc, the [Developer Guide](https://github.com/neomatrix369/nlp_profiler/blob/master/developer-guide.md) is also something to review as a closing action.

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (neomatrix369) [reply nlp_profiler/high_level_features/grammar_quality_check.py:1]: See https://pypi.org/project/black/ to install

Comment (bitanb1999) [reply nlp_profiler/high_level_features/grammar_quality_check.py:2]: hello 

> 

hello i have added comments and cleaned the code a bit more

File: requirements.txt
Comment (bitanb1999) [reply requirements.txt:14]: Yes, so previously, this happened with TextBlob, where their translate function stopped working due to the version upgrade. So, a safe way to go about it is, i believe, to fix the gingerit version to 0.9.0 unless it becomes obsolete.

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (neomatrix369) [reply nlp_profiler/high_level_features/grammar_quality_check.py:29]: You haven't applied this? Is sourcery able to parse the above line?

**Comment by neomatrix369** [comment]
@bitanb1999 Please read this comment and try to see if you can resolve it https://github.com/neomatrix369/nlp_profiler/pull/69#issuecomment-1463759805 - in either case respond on the PR with your findings

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (bitanb1999) [reply nlp_profiler/high_level_features/grammar_quality_check.py]: I have applied it shows on my end and have committed the same as well.

**Comment by bitanb1999** [comment]
> Please also do one last check in https://github.com/neomatrix369/nlp_profiler/blob/master/CONTRIBUTING.md to see if any dependent files need changing i.e. re-running notebooks etc, the [Developer Guide](https://github.com/neomatrix369/nlp_profiler/blob/master/developer-guide.md) is also something to review as a closing action.
> 
> Maybe you can enhance the existing grammar check example in the notebook(s) to illustrate the new package's features.
> 
> There are notebooks on this repo, please take a look at them and re-run them on your local machine to see if your changes have taken effect and no issues have arisen.
> 
> There are also markdown files in this repo, they may need a touch-up due to this change - can you pls check if that's the case?

I have gone through the contributing.md and I have abided by them all. Also, since the function is being just changed and there is no significant outer change in how the user will be calling the profiler, the notebooks remain as they are and the functions are to be called, as they were being called previously. Hence, no updates are needed in the notebooks.

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (neomatrix369) [reply nlp_profiler/high_level_features/grammar_quality_check.py:2]: https://www.freecodecamp.org/news/auto-format-your-python-code-with-black/

**Comment by neomatrix369** [comment]
If you see code format across the changes is inconsistent - linter/formatter would pick this up

File: nlp_profiler/high_level_features/grammar_quality_check.py
Comment (neomatrix369) [reply nlp_profiler/high_level_features/grammar_quality_check.py]: Push it and it will appear here - you may have applied to another branch on #71 but it hasn't reflected on this on

**Comment by neomatrix369** [comment]
One last thing to do is update the [CHANGELOG.md](https://github.com/neomatrix369/nlp_profiler/blob/master/CHANGELOG.md) for this change - its very easy to do, see how the previous ones are done

File: CHANGELOG.md
Comment (neomatrix369) [inline CHANGELOG.md:204]: Small fix: put today's date, you should see it in your commit info when you do `git show [commit hash]`

**Pull request merged**
